File: Design-Patterns/StateMachine/CSharp/StateMachine/Scenes/BattleScene.cs
using System;

namespace StateMachine.Scenes
{
    class BattleScene : Scene
    {
        public BattleScene(SceneController controller) : base(controller)
        {

        }

        public override void Start()
        {
            Console.WriteLine("Battle scene started");
        }

        public override void Update()
        {
            Console.WriteLine("Battle scene updated");
        }

        public override void End()
        {
            Console.WriteLine("Battle scene ended");
        }
    }
}

File: Design-Patterns/StateMachine/Cpp/StateMachine/battle_scene.cpp
#include "scene.h"

battle_scene::battle_scene(scene_controller* controller) : scene(controller)
{
	
}

void battle_scene::start()
{
	std::cout << "Battle scene started" << std::endl;
}

void battle_scene::update()
{
	std::cout << "Battle scene updated" << std::endl;
}

void battle_scene::end()
{
	std::cout << "Battle scene ended" << std::endl;
}
